### ISSUE TITLE
Settings should be written into registry or AppData folder

### ISSUE THREAD

**Issue by ralfbergs** [LibreHardwareMonitor/LibreHardwareMonitor#995]
I learnt in another issue the following:

> settings related to tray are written to "LibreHardwareMonitor.config" in the app folder 

I would argue this is very bad practice. The "Program Files" folder should be regarded as (and often **is**) read-only

Settings should better be written into the registry or stored into the AppData folder.

**Comment by DjordjeMandic** [comment]
There is no installer for the program, I don't see the point of having settings stored and accessible from any location. 

Maybe if there was installer which would install the program it would be really good practice to store settings in AppData. I don't like storing app settings in registry.

For now I think that this current approach is good.

If someone is willing to implement installer for LHM I'm then supporting the idea of storing settings in AppData.

I have implemented that in [my old program](https://github.com/DjordjeMandic/LeagueFPSBoost) and it stores it in user.config in appdata. Pretty simple to implement.

But still pointless without installer if app is going to be run from different location quite often.

Maybe I'm wrong :)

**Comment by ralfbergs** [comment]
Just to elaborate (and in part repeat myself): I suppose the "norm" for most users is to install LibreHardwareMonitor somewhere under `c:\Program Files\`, and not under their own user profile directory (`c:\users\<user>\`). This is a location that cannot be written to with normal user rights, it requires elevation. So it's a no-no for a program to write there.

Conclusion: settings should be stored in `c:\users\<user>\AppData\` or in the registry.

Of course, if you install the program under `c:\users\<user>\AppData\local\Programs\` then I can see why you would disagree, but IMHO most people won't store it there.

**Comment by ralfbergs** [comment]
I'm not a Windows developer, but I believe it should be stored in local, considering that roaming data seems [no longer supported in Windows 11](https://learn.microsoft.com/en-us/windows/apps/design/app-settings/store-and-retrieve-app-data). But again, I'm clueless.

**Comment by PhyxionNL** [comment]
The thing is that LHM doesn't have an installer, it's just a portable/zipped program. If it would have an installer, it would make sense to write it to %LocalAppData%, but _right now_ I don't think that writing it to that folder is what's always wanted.

**Comment by ks202005** [comment]
> I would argue this is very bad practice. The "Program Files" folder should be regarded as (and often is) read-only

This may have caused a lot of issues for me when configuring the program (at least on the original OpenHardwareMonitor). I had to troubleshoot multiple times why settings wouldn't be saved, though sometimes it was caused by not gracefully exiting (including graceful system shutdown). 

Now though, I install to my own software directory that shouldn't be read-only because of related issues with using ProgramFiles for portable software. 

I'd argue that LibreHardwareMonitor should use AppData by default (and make it obvious to users that's where configuration settings are stored), but allow a local .config file for portability.

**Comment by pritster5** [comment]
Is anyone having issues on the latest version with the settings not being saved after PC restart? all the sensors I hid or whatever sensor I chose to show in tray is being reset upon reboot :(

**Comment by tomlut** [comment]
Yes. Spent a while setting up the desktop widget. Restart -> all gone.

**Comment by AJolly** [comment]
I'd argue that settings should be stored in the same folder as the exe unless it's been installed.  Alternatively I have seen some programs write settings to AppData unless a portable.txt file is in the same directory.  Or have a config file that points to where settings should be stored.  

There is no reason the program should assume it will be installed in a folder it does not have write access to  (unless its been expressly put there via an elevated installer)

**Comment by bandaid** [comment]
Settings ought to remain in root directory of exe, as it is not an installer. Also, LOG files ought to be more manageable and remain in the root under Logs/YYYYMM__/


## Feature: Log Files Organized by Year-Month Folders

### Overview

This feature enhancement proposes organizing LibreHardwareMonitor’s log files into date-based folders by **year and month**, simplifying log management and reducing file clutter.

### Folder Structure Example

Log files will be saved in a folder named using the pattern:

```
Logs/YYYYMM__/
```

For example:

* Logs for June 2025 will be saved in:

  ```
  Logs/202506__/
  ```
* Logs for July 2025 will be saved in:

  ```
  Logs/202507__/
  ```

The day is not part of the folder name because it is included in the log file name itself.

### Implementation Details

* The log folder is dynamically generated based on the current system date when the log file is created.
* The code uses `DateTime.Now.ToString("yyyyMM")` to generate the folder name in the format `YYYYMM__`.
* The folder is created (if it doesn't already exist) at:

  ```
  [ApplicationRoot]/Logs/YYYYMM__/
  ```
* Existing log generation logic is updated to place the log file in the appropriate folder.

Example code snippet:

```csharp
string logRoot = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "Logs");
string dateFolder = DateTime.Now.ToString("yyyyMM") + "__";
string fullLogPath = Path.Combine(logRoot, dateFolder);
Directory.CreateDirectory(fullLogPath);

string logFileName = "LibreHardwareMonitor_" + DateTime.Now.ToString("yyyyMMdd_HHmmss") + ".csv";
string logFilePath = Path.Combine(fullLogPath, logFileName);
```

### Benefits

* Easier log organization by month.
* Prevents clutter in the main Logs directory.
* Simplifies log archival and cleanup.

### No Breaking Changes

This feature is backward-compatible:

* Existing log behavior remains unchanged for users not relying on specific log paths.
* Logs are grouped into month-based folders without impacting log content or format.

### Potential Next Steps

* Allow user customization of the log folder structure via settings (future improvement).
* Add a retention policy for old logs (future improvement).
